### PR TITLE
feat(kafka): add datastreams monitoring support for kafkajs batches

### PR DESCRIPTION
**Pull request by wconti27**
### What does this PR do?
Adds dsm checkpoints for kafkajs consumes using `eachBatch`

### Motivation
customer request

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes

Closes #3400

### CODE REVIEW

File: packages/datadog-instrumentations/src/kafkajs.js [hunk 1]
```diff
@@ -103,39 +107,72 @@ addHook({ name: 'kafkajs', file: 'src/index.js', versions: ['>=1.4'] }, (BaseKaf
...
-            } catch (e) {
-              consumerErrorCh.publish(e)
-              consumerFinishCh.publish(undefined)
-              throw e
+            })
+          }
+          : eachMessage,
+        eachBatch:
+          typeof eachBatch === 'function'
+            ? function (...eachBatchArgs) {
```
Comment (tlhunter) [inline packages/datadog-instrumentations/src/kafkajs.js:145]: This function and the one on line 113 are super similar. Can you instead come up with a common function to DRY the code?

File: packages/datadog-plugin-kafkajs/src/batch-consumer.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      for (const message of messages) {
+        if (message?.headers && DsmPathwayCodec.contextExists(message.headers)) {
+          const payloadSize = getMessageSize(message)
+          this.tracer.decodeDataStreamsContext(message.headers)
+          this.tracer
+            .setCheckpoint(['direction:in', `group:${groupId}`, `topic:${topic}`, 'type:kafka'], null, payloadSize)
+        }
+      }
+    }
+  }
```
Comment (tlhunter) [inline packages/datadog-plugin-kafkajs/src/batch-consumer.js:20]: Could remove two levels of indentation:

```suggestion
  start ({ topic, partition, messages, groupId }) {
    if (!this.config.dsmEnabled) return
    for (const message of messages) {
      if (!message || !message.headers || !DsmPathwayCodec.contextExists(message.headers)) continue
      const payloadSize = getMessageSize(message)
      this.tracer.decodeDataStreamsContext(message.headers)
      this.tracer
        .setCheckpoint(['direction:in', `group:${groupId}`, `topic:${topic}`, 'type:kafka'], null, payloadSize)
    }
  }
```

File: packages/datadog-instrumentations/src/kafkajs.js (hunk 1 above)
Comment (wconti27) [reply packages/datadog-instrumentations/src/kafkajs.js:145]: Updating this to use abstracted code to remove any duplication 👍

File: packages/datadog-plugin-kafkajs/src/batch-consumer.js (hunk 2 above)
Comment (wconti27) [reply packages/datadog-plugin-kafkajs/src/batch-consumer.js:20]: done

**Review by Qard — APPROVED** [review]

**Review by tlhunter — APPROVED** [review]

**Review by tlhunter — APPROVED** [review]

**Pull request merged**